### DISCUSSION TITLE
I cannot use a package named 'nutpie' in a professional setting, no matter how fast it is.

### DISCUSSION THREAD

**Discussion by alextsfox** [pymc-devs/nutpie#76] [General]
Sorry, but somebody had to say it; we were all thinking it. I'm sort of joking, but also not really.

**Comment by fonnesbeck** [comment]
Probably best to avoid "Google" and "Hugging Face" then! ;-)

**Comment by aseyboldt** [comment]
I'm not sure what you are referring to but as a non-native speaker I might be missing something. Do you mind telling me what this is about?

The name is really just because it implements NUTS (the no-u-turn sampler), is written in python, and because I like cake.

**Reply by alextsfox** [reply]
Hey, thanks for the sincere response! I understand that the etymology of the name comes from  the NUTS, the "No U-Turn Sampler,"  and that nut-filled pie is a real type of food (in the US, pecan pie is a very popular dessert, for example), so I didn't want to assume any malicious intent. However, "nutpie" (especially spelled as such) would be interpreted as something very different and extremely crude in American slang. I'm not crazy either, I asked around in my workplace for opinions on this, and everyone seemed to share the same opinion as myself. If you're curious as to a more in-depth explanation of why, I can dm you.

While my inner teenager does get a decent chuckle out of this, I would feel uncomfortable citing "nutpie" in an professional 
 or academic setting.

**Reply by ricardoV94** [reply]
I can't find any google search result for NSFW nutpie, even with safe search off. Urban dictionary also doesn't mention anything: https://www.urbandictionary.com/define.php?term=nut%20pie even though it's happy to describe what a creampie or sausage-fest is. Are you sure this is a widespread thing xD?

Sure it has nut in it, but so does nutella :D

**Reply by alextsfox** [reply]
that's fair, I'll drop it then -- I am just one person after all, and perhaps the meaning is not as obvious as I thought (it may in fact be a more regional+generational thing).

**Comment by mike-lawrence** [comment]
Possibly a generational/regional thing. I’m an xennial/elder-millennial from Canada and the name was entirely innocuous to me. Once OP brought it up, yes I saw that it could be read as lewd, but I think it’s a stretch to say it’s obvious. But if it’s really an issue, maybe the maintainer would consider pynuts (could have an icon of a pine nut). 

For those in the dark, “nut” can be used as slang in a sexual context, in which case “nut pie” could be seen as a synonym for “cream pie”, which is also used as slang in a sexual context.

**Comment by twiecki** [comment]
Hate to be that guy, but after having made this association I now wince whenever someone says the name of the package :(. Maybe we could just brainstorm some alternatives and see if we find something we like?

I like @mike-lawrence `pynuts` for example, `fastnuts`.

**Comment by canyon289** [comment]
I can see the implication, I get that in a professional setting there's a small chance someone might think its bad. The issue I have is thats its putting an a big burden on the maintainers to rename this, and all the downstream content that needs to be changed. And then there's the issue that it becomes obvious this package was renamed, and people asking and wondering about it across all code bases.

FWIW I went to common sites for the type of material this might be associated with. The search team nutpie did not bring up any content, where as other common slang terms that are also innocuous did.

I don't think it's worth changing the package name

**Reply by twiecki** [reply]
I don't think it'd be that much work to change it, and the package is still fairly unknown.

**Reply by ricardoV94** [reply]
I still disagree with changing. PyNuts / FastNuts has much more obvious connotations for me.. It's going to be highly subjective and a slippery slope.

**Reply by canyon289** [reply]
To be precise I'll support either decision. Delegating any voting power I'd have to the folks closer to the package!

**Comment by trendelkampschroer** [comment]
First of all I think nutpie is a great project. Thanks a lot for creating and making it available.

To me the name did not (until now) evoke any unwanted/problematic associations. I am a non native speaker though.

 I guess we're using std all the time, then there is the sandwich estimator...

If you want to find something I guess you will. 

I'd not bend package names too much in order to avoid any far fetched association.

**Comment by maresb** [comment]
Ya, by now I think that the maintainers have better things to do than worry about potential innuendos. And regarding a "professional" setting, part of being professional is to be able to power through awkward situations.

I suggest we close this discussion because I don't think we will take action, and the increased visibility from having it open is just increasing the number of people who discover the innuendo.
